### ISSUE TITLE
[BUG] Pretty-code with ligature font flag crashes client

### ISSUE THREAD

**Issue by chadyehia** [hlissner/doom-emacs#2169]
**Describe the issue**
Adding font ligature support to the pretty-code module (e.g. `(pretty-code +hasklig)`) crashes Emacs on startup. I ran `doom refresh` beforehand. I removed the flag and Emacs ran normally.

NOTE: This works normally when running Emacs in daemon mode.

- What did you expect to happen?

Emacs starts up normally, font ligatures are displayed.

- What actually happened?

Frame pops up for a split second then crashes.

<details>

<pre>
❯ emacs --debug-init .
Fatal error 11: Segmentation faultfish: 'emacs --debug-init .' terminated by signal SIGABRT (Abort)
</pre>
</details>

**Steps to reproduce**
1. Download any ligature-enabled font (I tried Hasklig)
2. Set doom-font to Hasklig
3. Set +haslig flag to the pretty-code module
4. Exit and doom refresh.
5. Try to start Emacs.

**System information**
<details><pre>
((emacs
  (version . "26.3")
  (features . "JPEG RSVG IMAGEMAGICK GLIB NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2")
  (build . "Nov 17, 2019")
  (buildopts "--disable-dependency-tracking --disable-silent-rules --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/local/Cellar/emacs-plus/26.3/share/info/emacs --prefix=/usr/local/Cellar/emacs-plus/26.3 --with-xml2 --without-dbus --with-gnutls --with-imagemagick --with-modules --with-rsvg --with-ns --disable-ns-self-contained")
  (windowsys . batch)
  (daemonp))
 (doom
  (version . "2.0.9")
  (build . "HEAD -> develop, origin/develop, origin/HEAD 385ff35ad 2019-12-08 01:02:01 -0500"))
 (system
  (type . darwin)
  (config . "x86_64-apple-darwin19.0.0")
  (shell . "/usr/local/bin/fish")
  (uname . "Darwin 19.0.0 Darwin Kernel Version 19.0.0: Thu Oct 17 16:17:15 PDT 2019; root:xnu-6153.41.3~29/RELEASE_X86_64 x86_64")
  (path "/usr/local/bin" "/usr/bin" "/bin" "/usr/sbin" "/sbin" "~/.emacs.d/bin" "/usr/local/lib/ruby/gems/2.6.0/bin" "/usr/local/Cellar/emacs-plus/26.3/libexec/emacs/26.3/x86_64-apple-darwin19.0.0"))
 (config
  (envfile . envvar-file)
  (elc-files . 0)
  (modules :completion company ivy :ui deft doom doom-dashboard doom-quit fill-column hl-todo modeline nav-flash ophints (popup +all +defaults) (pretty-code +hasklig) unicode vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold multiple-cursors parinfer rotate-text snippets word-wrap :emacs dired electric ibuffer vc :term eshell :tools debugger docker editorconfig (eval +overlay) flycheck flyspell (lookup +docsets) lsp macos magit rgb :lang data emacs-lisp javascript ledger markdown (org +dragndrop +ipython +pandoc +pomodoro +present) python rest ruby (sh +fish) web :app calendar write :config (default +bindings +smartparens))
  (packages ob-mongo ob-http "(solaire-mode :disable t)")
  (elpa "n/a")))
</pre></details>

**Comment by chadyehia** [comment]
As a workaround, I copied the content of the +hasklig.el file into my private config.el, and it seems to be working.

**Comment by hlissner** [comment]
This is likely all you needed.

```elisp
(defvar +pretty-code-hasklig-font-ligatures
  '(("&&"   . #Xe100)
    ("***"  . #Xe101)
    ("*>"   . #Xe102)
    ("\\\\" . #Xe103)
    ("||"   . #Xe104)
    ("|>"   . #Xe105)
    ("::"   . #Xe106)
    ("=="   . #Xe107)
    ("==="  . #Xe108)
    ("==>"  . #Xe109)
    ("=>"   . #Xe10a)
    ("=<<"  . #Xe10b)
    ("!!"   . #Xe10c)
    (">>"   . #Xe10d)
    (">>="  . #Xe10e)
    (">>>"  . #Xe10f)
    (">>-"  . #Xe110)
    (">-"   . #Xe111)
    ("->"   . #Xe112)
    ("-<"   . #Xe113)
    ("-<<"  . #Xe114)
    ("<*"   . #Xe115)
    ("<*>"  . #Xe116)
    ("<|"   . #Xe117)
    ("<|>"  . #Xe118)
    ("<$>"  . #Xe119)
    ("<>"   . #Xe11a)
    ("<-"   . #Xe11b)
    ("<<"   . #Xe11c)
    ("<<<"  . #Xe11d)
    ("<+>"  . #Xe11e)
    (".."   . #Xe11f)
    ("..."  . #Xe120)
    ("++"   . #Xe121)
    ("+++"  . #Xe122)
    ("/="   . #Xe123)
    (":::"  . #Xe124)
    (">=>"  . #Xe125)
    ("->>"  . #Xe126)
    ("<=>"  . #Xe127)
    ("<=<"  . #Xe128)
    ("<->"  . #Xe129)))

(set-fontset-font t '(#Xe100 . #Xe129) +pretty-code-hasklig-font-name)
```

If you're using the `emacs-mac` formula, which has native support for ligatures (iirc), then there's like some conflict. I'm not a mac or ligature user so I don't know how best to resolve this.

**Comment by chadyehia** [comment]
It's this line that's causing the issue for me:
```
(set-fontset-font t '(#Xe100 . #Xe129) +pretty-code-hasklig-font-name)
```

If I comment it out in the module itself, everything works fine. I'm still an elisp newb so I don't want to submit a PR yet for a fix, because frankly I have no idea what that line does lol!

**Comment by hlissner** [comment]
That line maps a range of characters in the unicode space of the "current" font to the characters in the Hasklig font.

Are you sure you have the [hasklig](https://github.com/i-tu/Hasklig/) font installed?

**Comment by chadyehia** [comment]
Yup, I just doubled checked, I have Hasklig and Fira Code installed.
